src/abstractdataframe/abstractdataframe.jl

```diff
@@ -983,35 +992,78 @@ julia> vcat(df1, df2)
+Base.vcat(df::AbstractDataFrame; 
+          columns::Union{Symbol, AbstractVector{Symbol}}=:equal) = df
+
+Base.vcat(dfs::AbstractDataFrame...;
+          columns::Union{Symbol, AbstractVector{Symbol}}=:equal) = 
+    _vcat(collect(dfs); columns=columns)
+
+function _vcat(dfs::AbstractVector{<:AbstractDataFrame}; 
+               columns::Union{Symbol, AbstractVector{Symbol}}=:equal)
     isempty(dfs) && return DataFrame()
+    # Array of all headers
     allheaders = map(names, dfs)
+    # Array of unique headers across all data frames
     uniqueheaders = unique(allheaders)
+    # All symbols present across all headers
     unionunique = union(uniqueheaders...)
+    # List of symbols present in all dataframes
     intersectunique = intersect(uniqueheaders...)
-    coldiff = setdiff(unionunique, intersectunique)
-
-    if !isempty(coldiff)
-        # if any DataFrames are a full superset of names, skip them
-        filter!(u -> Set(u) != Set(unionunique), uniqueheaders)
-        estrings = Vector{String}(undef, length(uniqueheaders))
-        for (i, u) in enumerate(uniqueheaders)
-            matching = findall(h -> u == h, allheaders)
-            headerdiff = setdiff(coldiff, u)
-            cols = join(headerdiff, ", ", " and ")
-            args = join(matching, ", ", " and ")
-            estrings[i] = "column(s) $cols are missing from argument(s) $args"
+
+    # Throw an informative error if names arent the same across all dataframes
+    if columns === :equal
+            header = unionunique
+            coldiff = setdiff(unionunique, intersectunique)
+        if !isempty(coldiff) 
+            # if any DataFrames are a full superset of names, skip them
+            filter!(u -> Set(u) != Set(header), uniqueheaders)
+            # string to hold all the errors, one for each missing column name
+            @show uniqueheaders
+            estrings = Vector{String}(undef, length(uniqueheaders))
+            for (i, head) in enumerate(uniqueheaders)
+                # Find all the headers that match the header you are on. 
+                # So that we don't repeat ourselves in the error.
+                matching = findall(h -> head == h, allheaders)
+                # Of the symbols that aren't in all headers, which 
+                # ones are in this one?
+                headerdiff = setdiff(coldiff, head)
+                cols = join(headerdiff, ", ", " and ")
+                args = join(matching, ", ", " and ")
+                estrings[i] = "Column(s) $cols are missing from argument(s) $args"
+            end
+            throw(ArgumentError(join(estrings, ". ")))
         end
-        throw(ArgumentError(join(estrings, ", ", ", and ")))
+    # Only have columns that are in all dataframes. No error thrown. 
+    elseif columns === :intersect 
+        header = intersectunique
+    elseif columns === :union
+        header = unionunique
+        # no setdiff cause we take in everything
```

Comment:
These comments look out of place. Same above. Avoid describing the behavior already described in the docstring.

Reply:
Sounds good. I might keep a few in while I work. Do you mind deleting the unnecessary ones at the end?